src/Expect.elm

```diff
@@ -813,7 +813,7 @@ withinCompare tolerance a b =
             a - absolute tolerance <= b && b <= a + absolute tolerance
 
         withinRelativeTolerance =
-            (a * (1 - relative tolerance) <= b && b <= a * (1 + relative tolerance))
-                || (b * (1 - relative tolerance) <= a && a <= b * (1 + relative tolerance))
+            (a  - (abs (a * relative tolerance)) <= b && b <= a + (abs (a * relative tolerance)))
+                || (b - (abs (b * relative tolerance)) <= a && a <= b + (abs (b * relative tolerance)))
     in
-    (a == b) || withinAbsoluteTolerance || withinRelativeTolerance
+        (a == b) || withinAbsoluteTolerance || withinRelativeTolerance
```

Comment:
I'm not entirely convinced that just adding `abs` does the correct thing here, e.g. when only one argument is negative. It looks fine, but I'd like some tests to be sure.

Reply:
Think of a number line. Take your nominal value(n) and and the absolute value of the tolerance(t). Then check if the actual value is between n-t and n+t.  This is the way that absolute worked from the beginning and it is very simple and robust.  Relative was using (1-t)*n, which works for simple cases, but gets all weird when n is negative or t is greater than one because it moves the tolerance band in the opposite direction.  All this is to say that I didn't just add 'abs', but changed the algorithm to eliminate the source of the bug.  By the way, my initial solution was to use a ` case` statement to check if the nominal value was less then zero and then use a branch where all the comparisons where flipped around to work with the inversion caused by the negative values, but this caused the communatative tests to fail - fuzz was send all kinds of weird values, like 0 for nominal, or tolerance greater than 1, which triggered all the edge-case failure modes in the original approach.  This led me to my current solution.

Reply:
I hope this answers your question below as well about having a similar issue in absolute.  I did tests for Absolute tolerances as well, and they all passed without changes.